crates/block-producer/src/domain/batch.rs

```diff
@@ -0,0 +1,104 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use miden_objects::Word;
+use miden_objects::account::AccountId;
+use miden_objects::batch::BatchId;
+use miden_objects::transaction::TransactionId;
+
+use crate::domain::transaction::AuthenticatedTransaction;
+
+// SELECTED BATCH
+// ================================================================================================
+
+/// A sequence of transactions selected by the [`Mempool`] to be processed by the
+/// [`BatchBuilder`] into a [`ProposedBatch`], and then finally into a [`ProvenBatch`].
+///
+/// [Mempool]: crate::mempool::Mempool
+/// [BatchBuilder]: crate::batch_builder::BatchBuilder
+/// [ProposedBatch]: miden_objects::batch::ProposedBatch
+/// [ProvenBatch]: miden_objects::batch::ProvenBatch
+#[derive(Clone, Debug, PartialEq)]
+pub(crate) struct SelectedBatch {
+    txs: Vec<Arc<AuthenticatedTransaction>>,
+    id: BatchId,
+    account_updates: HashMap<AccountId, (Word, Word)>,
+}
+
+impl SelectedBatch {
+    pub(crate) fn builder() -> SelectedBatchBuilder {
+        SelectedBatchBuilder::default()
+    }
+
+    pub(crate) fn id(&self) -> BatchId {
+        self.id
+    }
+
+    pub(crate) fn txs(&self) -> &[Arc<AuthenticatedTransaction>] {
+        &self.txs
+    }
+
+    pub(crate) fn into_transactions(self) -> Vec<Arc<AuthenticatedTransaction>> {
+        self.txs
+    }
+
+    pub(crate) fn account_updates(&self) -> impl Iterator<Item = (AccountId, Word, Word)> {
```

Comment:
nit: I think it's fair to assume that anyone reading this can easily infer the return type, but I'd maybe still mention it in a doc comment (or maybe at the level of `SelectedBatch`)